### ISSUE TITLE
transforms are published even if disabled

### ISSUE THREAD

**Issue by christian-rauch** [ros-planning/moveit_calibration#59]
The plugin seems to publish to `/tf`, event when it is disabled, i.e. when it is unchecked in the Display overview. This is very dangerous as it will override transforms published by a robot state publisher, once RViz starts. I would expect that the calibrated transform is only published once a calibration procedure has finished.

**Comment by JStech** [comment]
Just want to make sure, you mean this checkbox here?
![tf_display](https://user-images.githubusercontent.com/519268/104941410-4132f580-5970-11eb-8c4e-3ac83e4c9b8e.png)

That doesn't disable anything besides the display of `/tf` in RViz (frames and transforms).

The transform that the plugin publishes is the transform that is being calibrated. In my experience, that transform isn't already being published anywhere, but if you already have a nominal transform that you want to start with, you can set it as the initial guess on the context tab. I suppose when the frames are selected it could check if the transform is already being published and load it into the initial guess.

**Comment by christian-rauch** [comment]
@JStech Yes, this is the checkbox I meant. Unchecking it will hide the plugin from the 3D view and from the panel view. This suggests that the plugin is not active anymore. Nevertheless, it is still publishing transforms. You actually have to unload the plugin to stop it from publishing transforms.

It makes sense for the plugin to pickup transformations they are already published, and to publish the solution once a calibration was successful, but it should not override already published transforms, when no calibration is going on.

**Comment by JStech** [comment]
Yeah, I've hit a similar issue by added the panel multiple times, which causes other problems (#12).

If you want to disable the transform publishing, it's probably best to un-set the frames in the context tab or delete the calibration panel entirely. I think it's reasonable to assume that if it's loaded it's running.

**Comment by christian-rauch** [comment]
Well, "running" could be interpreted differently :-) For me, "running" only applies to the phase where you actually do the calibration, i.e. have the solver running once on the gathered samples.

I think it is very dangerous to override the transforms that are published by the robot state publisher. The plugin is doing this silently when it appears to be inactive as it does not any traces in the UI.

I also do not see the benefit of publishing the calibrated transformation after RViz has been restarted.
The procedure should be:
1. calibrate once,
2. have the plugin publish the transform to verify it is correct in RViz,
3. save the transformation
4. the user has to integrate the stored transformation into the robot launch script or the robot description.

Relying on RViz to "restore" the calibration is not reliable as you may want to use a robot without RViz or load a new RViz configuration.
